crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -95,7 +95,27 @@ impl StringFunction {
             Extract { .. } => mapper.with_same_dtype(),
             ExtractAll => mapper.with_dtype(DataType::List(Box::new(DataType::Utf8))),
             #[cfg(feature = "extract_groups")]
-            ExtractGroups { .. } => mapper.with_same_dtype(),
+            ExtractGroups { pat } => {
```

Comment:
Can we do this in the expression function and then put the DataType in the enum variant?

We can use the datatype in the expression implementation to get the names.

Reply:
Is there an example I can refer to in order to try and understand what that means?

It's above my current rust knowledge level.